hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieTableServiceClient.java

```diff
@@ -919,17 +919,26 @@ protected Map<String, Option<HoodiePendingRollbackInfo>> getPendingRollbackInfos
    */
   protected boolean rollbackFailedIndexingCommits() {
     HoodieTable table = createTable(config, hadoopConf);
-    List<String> instantsToRollback = getFailedIndexingCommitsToRollback(table.getMetaClient());
+    List<String> instantsToRollback = getFailedIndexingCommitsToRollbackForMetadataTable(table.getMetaClient());
     Map<String, Option<HoodiePendingRollbackInfo>> pendingRollbacks = getPendingRollbackInfos(table.getMetaClient());
     instantsToRollback.forEach(entry -> pendingRollbacks.putIfAbsent(entry, Option.empty()));
     rollbackFailedWrites(pendingRollbacks);
     return !pendingRollbacks.isEmpty();
   }
 
-  protected List<String> getFailedIndexingCommitsToRollback(HoodieTableMetaClient metaClient) {
+  private List<String> getFailedIndexingCommitsToRollbackForMetadataTable(HoodieTableMetaClient metaClient) {
+    if (!isMetadataTable(metaClient.getBasePathV2().toString())) {
+      return Collections.emptyList();
+    }
+    HoodieTableMetaClient dataMetaClient = HoodieTableMetaClient.builder()
+        .setBasePath(HoodieTableMetadata.getDatasetBasePath(config.getBasePath()))
+        .setConf(metaClient.getHadoopConf())
+        .build();
+    HoodieTimeline dataIndexTimeline = dataMetaClient.getActiveTimeline().filter(instant -> instant.getAction().equals(HoodieTimeline.INDEXING_ACTION));
+
```

Comment:
L939 to 943: why can't we introduce getIndexingTimeline to DefaultTimeline similar to getCommitsTimeline.

Reply:
I think we could.